### ISSUE TITLE
DNS and Tor problems

### ISSUE THREAD

**Issue by umbrageous** [jlund/streisand#275]
As you will see, I am a bit of a newbie to all of this!

I installed Streisand on a new Digital Ocean droplet without any problems [via Linux Mint 17.3 running in a VirtualBox VM]. All the associated OpenVPN, stunnel and obfs3 also, on my Win7 SP1 host machine.

OpenVPN and OpenVPN/stunnel connect fine on Windows. But, when I try to browse to, say, ipleak.net to check IP address and DNS resolver(s), the browser will not connect, and I get DNS_PROBE_FINISHED_BAD_CONFIG.

Also, when I place the obfs3 custom bridge in the right place in the Tor Browser Bundle configuration (again in Windows), connection simply hangs. I know (hidden) bridges work on TBB, because I can connect Tor using them, so it's not a TBB problem.

Before installing Streisand, I have generally used a commercial VPN to connect everything to the web [AirVPN, using its OpenVPN GUI and its Network Lock facility], but I ensure that that is closed, logged out, and Network Lock de-activated before starting Streisand OpenVPN etc. The fact that I get a good OpenVPN or OVPN/stunnel connection suggests to me that this is not interfering in any way.

I also have tended to use DNSCrypt to encrypt DNS when connecting via non-Streisand VPN, but not as a service, and I have checked that it is inactive and all DNS adapters are marked as 'manual' when connecting through OVPN etc to Streisand.

Can you advise as to what I should do about this DNS resolver problem? Please bear in mind that I know how to connect via Linux to the Streisand server by using ssh root@<ip-address>, but am hazy on commands to use so as to managing settings thereafter.

I am not in a location where internet connections are restricted.

Any help would be appreciated. Thanks.

**Comment by umbrageous** [comment]
I should add that the bridge QR code does not seem to work for Orbot, either. "Use Bridges" not ticked, and no bridge in the bridges folder.

**Comment by umbrageous** [comment]
I have solved this. It was a conflict with the AirVPN interface. Shutting that interface down was not enough. Once I then flushed DNS back to default, both OpenVPN and stunnel worked fine, and DNS resolved. So I can use both on the same PC, it seems.

I spun up another droplet, and the obfs3 bridge works fine on both TBB and Orbot - though you may wish to note that Orbot still would not work with the QR code provided.

**Comment by jlund** [comment]
Cool! I'm glad that you found a solution to the main problem.

As for the Orbot QR code, that's still working for me. What version of Android are you using?

**Comment by umbrageous** [comment]
Thanks for the response. Marshmallow.

**Comment by jlund** [comment]
Darn. My test device is Lollipop, and the QR code is still working for me. I will see what I can figure out.

**Comment by jlund** [comment]
I think I have this Orbot QR issues fixed in the Xenial branch (that also has obfs4 support). I'm trying to get it merged into master today :)
